### DISCUSSION TITLE
Only create updates that are listed in packageRules

### DISCUSSION THREAD

**Discussion by ghost** [renovatebot/renovate#15964] [Request Help]
We are currently starting to use renovate and so far it has been great :D. The only problem we are struggling with is to prevent all updates except the packages which match the packageRules.

Our config:

```
{
    "schedule": [
        "after 8am every weekday",
        "before 3pm every weekday"
    ],
    "branchPrefix": "renovate/",
    "prConcurrentLimit": 5,
    "labels": ["dependencies"],
    "baseBranches": ["develop", "master", "main"],
    "separateMultipleMajor": true,
    "platformAutomerge": true,
    "separateMinorPatch": true,
    "packageRules": [
        {
            "matchUpdateTypes": ["patch", "minor"],
            "matchBaseBranches": ["master", "main"],
            "matchCurrentVersion": "!/^0/",
            "groupName": "Master - Patch/Minor",
            "enabled": true,
            "prPriority": 5,
            "automerge": true,
            "automergeType": "pr",
            "labels": ["master-dependency-updates"]
        },
        {
            "matchUpdateTypes": ["major"],
            "matchBaseBranches": ["develop"],
            "automerge": false,
            "groupName": "Develop - Major",
            "enabled": true,
            "prPriority": 1,
            "labels": ["develop-major-updates"]
        }
    ],
    "reviewers": ["liberatys"],
    "lockFileMaintenance": {
        "enabled": true,
        "automerge": true
    },
    "timezone": "Europe/Zurich"
}
```

This will create the groups we want for our updates! But sadly it will also create  some other pull requests against develop/master that we do not want.

I tried adding an additional packageRule:

```
{
   "matchPackagePatterns": [
      "*"
   ],
   "enabled": false
}
```

but this will sadly just disable all updates. 

There is probably something in the documentation that would do what I want ^^ but after searching github issues / prs and the docs I still came up empty 🤷🏻.

Thank you for the great product!

**Comment by crh225** [comment]
Did the other branches get created before adding the * rule enabled false? 
If so, they will keep creating themselves: https://github.com/renovatebot/renovate/discussions/15954#discussioncomment-2910547
That rule looks correct to me.

**Comment by rarkins** [comment]
1. Put your disabling first, and your selective enabling second. Last matching rule wins, if they contradict
2. Remember that if you disable based on package name like in your example, Renovate won't even bother to look up the package because you disabled it. But if it doesn't look it up, then it won't get far enough to have any minor/patch/major classification, so your enabling rule won't work. Therefore it you want to enable using matchUpdateTypes, you need to disable using them too
3. Instead of disabling, I recommend you instead use `dependencyDashboardApproval=true` for anything you don't want, if you are on a supported platform like github or gitlab. That way you still have visibility into it.
